rdr_service/api/nph_participant_api_schemas/schema.py

```diff
@@ -0,0 +1,308 @@
+import logging
+from graphene import (
+    ObjectType,
+    String,
+    Int,
+    Date,
+    DateTime,
+    NonNull,
+    Field,
+    List,
+    Schema
+)
+from graphene import relay
+from sqlalchemy.orm import Query
+
+from rdr_service.model.participant import Participant as DbParticipant
+from rdr_service.model.nph_sample import NphSample
+from rdr_service.dao import database_factory
+from rdr_service.api.nph_participant_api_schemas.util import SortContext, loadParticipantData
+
+
+class SortableField(Field):
+    def __init__(self, *args, sort_modifier=None, **kwargs):
+        super(SortableField, self).__init__(*args, **kwargs)
+        self.sort_modifier = sort_modifier
+
+    @staticmethod
+    def sort(current_class, field_name, context):
+        return current_class.sort(context, field_name)
+
+
+class Event(ObjectType):
+    """ NPH Participant Event Status """
+
+    value = SortableField(
+        NonNull(String), sort_modifier=lambda context: context.set_order_expression(context.sort_table.status)
+    )
+    time = SortableField(
+        NonNull(DateTime),
+        sort_modifier=lambda context: context.set_order_expression(context.sort_table.time)  # Order by time
+    )
+
+    @staticmethod
+    def sort(context, value):
+        if value.upper() == "TIME":
+            return context.set_order_expression(context.sort_table.time)
+        return context.set_order_expression(context.sort_table.status)
+
+
+class EventCollection(ObjectType):
+    current = SortableField(Event)
+    # TODO: historical field need to sort by newest to oldest for a given aspect of a participant’s data
+    historical = List(Event)
+
+    @staticmethod
+    def sort(context, value):
+        pass
+
+
+class Sample(ObjectType):
+    parent = SortableField(EventCollection)
+    child = SortableField(EventCollection)  # todo: join 'child' on parent ('sample') and sort_table = 'child'
```

Comment:
This todo seems like it's already implemented below, was it left in by accident or is it still to remind of something?

Reply:
removed the TODO item